File: slides/code/01-cpp/IntCell.java
// File IntCell.java

public class IntCell {
    public IntCell() {
        this(0);
    }

    public IntCell(int initialValue) {
        storedValue = initialValue;
    }

    public int getValue() {
        return storedValue;
    }

    public void setValue(int val) {
        storedValue = val;
    }

    private int storedValue;

    public static void main( String [] args ) {
        IntCell m1 = new IntCell();
        IntCell m2 = new IntCell(37);
        System.out.println(m1.getValue( ) + " " + m2.getValue( ) );
    }

}

File: slides/code/01-cpp/IntCell.cpp
// File IntCell.cpp

#include "IntCell.h"
using namespace std; // (not really necessary, but...)

IntCell::IntCell( int initialValue ) : storedValue( initialValue ) {

}

int IntCell::getValue( ) const {
    return storedValue;
}

void IntCell::setValue( int val ) {
    storedValue = val;
}

int IntCell::max(int m) const{
    return (m>storedValue) ? m : storedValue;
}
